### ISSUE TITLE
Reserved keyword names should have double backticks

### ISSUE THREAD

**Issue by JordanMarr** [edgarfgp/Fabulous.AST#67]
Creating a name using a reserved keyword should have double backticks.
(Records, Values, Fields, Modules, etc).

Example:
```F#
Value("class", "1", false)
```
Should produce:
```F#
let ``class`` = 1
```

And the new failing test case :

`[<InlineData("class", "``class``")>]`

```F#
module RecordExpr =

    [<Theory>]
    [<InlineData("Red Blue", "``Red Blue``")>]
    [<InlineData("Red_Blue", "Red_Blue")>]
    [<InlineData(" Red Blue ", "``Red Blue``")>]
    [<InlineData("net6.0", "``net6.0``")>]
    [<InlineData(" net6.0 ", "``net6.0``")>]
    [<InlineData("class", "``class``")>]
    let ``Produces an AnonRecordExpr with fields with backticks`` (value: string) (expected: string) =
        AnonymousModule() { AnonRecordExpr() { RecordFieldExpr(value, ConstantExpr("1", false)) } }
        |> produces
            $$"""

```

**Comment by JordanMarr** [comment]
Here is one more failing test case to add for the names:

`[<InlineData("2013", "``2013``")>]`


For example, this is a generated table from the SqlHydra tests:

```F#
    [<CLIMutable>]
    type vSalespersonSalesByFiscalYears =
        { SalesPersonID: Option<int>
          FullName: Option<string>
          JobTitle: Option<string>
          SalesTerritory: Option<string>
          ``2012``: Option<decimal>
          ``2013``: Option<decimal>
          ``2014``: Option<decimal> }

```

**Comment by edgarfgp** [comment]
Agreed. I’m travelling next week but will try to look into this

**Comment by edgarfgp** [comment]
The relevant check is [here](https://github.com/edgarfgp/Fabulous.AST/blob/6fe1275c66dec5ccfe75dff8079aefdf8d884934/src/Fabulous.AST/Widgets/SingleTextNode.fs#L13) , we will would need to extend this active pattern

**Comment by JordanMarr** [comment]
I’m willing to give it a shot if you’re open to pull requests (unless that would be disruptive to your flow).

**Comment by edgarfgp** [comment]
> I’m willing to give it a shot if you’re open to pull requests (unless that would be disruptive to your flow).

Go for it 👍🏻. I do not have anything pending for now, just need to write docs and recipes.

**Comment by edgarfgp** [comment]
I guess we can cover the keyword-reference with backticks  https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/keyword-reference

**Comment by JordanMarr** [comment]
I was poking around and found the `PrettyNaming` module that has all these cool helper functions around naming. 
It looks like this might handle all the backtick cases!

```F#
/// Contains all the single text nodes that are used in the AST
[<RequireQualifiedAccess>]
module SingleTextNode =
    [<return: Struct>]
    let (|WrapWithBackTicks|_|) (str: string) =
        if System.String.IsNullOrEmpty str then
            failwith "This is not a valid identifier"
        else
            let str = str.Trim()
            let normalized = PrettyNaming.NormalizeIdentifierBackticks(str)
            if normalized <> str then
                ValueSome(SingleTextNode.Create(normalized))
            else
                ValueNone

```

**Comment by JordanMarr** [comment]
Now that I think about it, this could be streamlined by adding as a helper:

```F#
/// Adds double backticks to the identifier if necessary.
let normalizeIdentifierBackticks (identifier: string) = 
    if System.String.IsNullOrEmpty identifier then
        failwith "This is not a valid identifier"
    else
        let trimmed = identifier.Trim()
        Fantomas.FCS.Syntax.PrettyNaming.NormalizeIdentifierBackticks trimmed
```

And then we can just pipeline it where needed:

```F#
module RecordField =
    let RecordExpr = Attributes.defineWidget "RecordExpr"

    let Name = Attributes.defineScalar<string> "Name"

    let WidgetKey =
        Widgets.register "RecordField" (fun widget ->
            let expr = Helpers.getNodeFromWidget<Expr> widget RecordExpr
            let name = 
                Helpers.getScalarValue widget Name
                |> Helpers.normalizeIdentifierBackticks
                |> SingleTextNode.Create

            RecordFieldNode(
                IdentListNode([ IdentifierOrDot.Ident(name) ], Range.Zero),
                SingleTextNode.equals,
                expr,
                Range.Zero
            ))
```

**Comment by edgarfgp** [comment]
Fixed in #68 . It is available in `0.8.2`
